### PR TITLE
Remove nums

### PR DESCRIPTION
**Pull request by Halbaroth**
This PR removes the support for the deprecated Nums library as suggested in #590.

### CODE REVIEW

**Comment by bclement-ocp** [comment]
The parser and ast are still using num in this PR I believe, is that intentional?

**Comment by Halbaroth** [comment]
No, I didn't notice it... I modified the legacy parser to use only Zarith.

File: src/parsers/psmt2_to_alt_ergo.ml [hunk 1]
```diff
@@ -115,8 +115,8 @@ module Translate = struct
     in
     aux sort.ty
 
-  let better_num_of_string s =
-    begin match String.split_on_char '.' s with
+  (* -- unused -- let better_num_of_string s =
```
Comment (bclement-ocp) [inline src/parsers/psmt2_to_alt_ergo.ml]: Let's just remove this. It is in the git history if we really need it.

**Review by bclement-ocp — APPROVED** [review]
This looks good to me, please just remove the old code from the comments.

I believe `decimal_number` (and `hexa_number`) could be implemented with `Q.of_string` (`Q.of_string ("0x" ^ x)`, respectively), but the specialized implementation is probably a tiny bit faster. Not that it matters much…

File: src/parsers/psmt2_to_alt_ergo.ml (hunk 1 above)
Comment (Halbaroth) [reply src/parsers/psmt2_to_alt_ergo.ml]: Sure but we will remove the legacy parser so I did the minimal effort.

**Pull request merged**